### PR TITLE
Add !jsonDatasource

### PR DESCRIPTION
**Pull request by pvalsecc**

### CODE REVIEW

**Comment by pvalsecc** [comment]
If you have access, here is how it can be used: https://github.com/camptocamp/saccas_suissealpine/commit/d83ee0d412b2f5386b5bdaafcfab4cdea0649b1b

File: core/src/main/java/org/mapfish/print/attribute/JsonDatasourceAttribute.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            throw new IllegalArgumentException(message);
+        }
+
+        try {
+            final JsonDataSource result = new JsonDataSource(new ByteArrayInputStream(json.getBytes(StandardCharsets.UTF_8)));
+            result.next();
+            return result;
+        } catch (JRException e) {
+            throw new RuntimeException("Error while parsing " +
+                    requestJsonAttributes.getPath(attributeName) + "as json", e);
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/attribute/JsonDatasourceAttribute.java]: Missing space before as?

File: core/src/main/java/org/mapfish/print/output/Values.java [hunk 2]
```diff
@@ -89,32 +82,27 @@ public Values() {
 
     /**
      * Construct from the json request body and the associated template.
-     *
-     * @param jobId the job ID
+     *  @param jobId the job ID
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/output/Values.java]: Extra space :-)

File: core/src/main/java/org/mapfish/print/output/Values.java [hunk 3]
```diff
@@ -89,32 +82,27 @@ public Values() {
...
                   final File jasperTemplateBuild) throws JSONException {
-        this(jobId, requestData, template, parser, taskDirectory, httpRequestFactory, jasperTemplateBuild, null);
+        this(jobId, requestData, template, taskDirectory, httpRequestFactory, jasperTemplateBuild, null);
     }
 
     /**
      * Construct from the json request body and the associated template.
-     *
-     * @param jobId the job ID
+     *  @param jobId the job ID
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/output/Values.java]: Extra space :-)

File: core/src/main/java/org/mapfish/print/parser/MapfishParser.java [hunk 4]
```diff
@@ -134,10 +135,10 @@ public void parse(final boolean errorOnExtraProperties, final PObject requestDat
         }
     }
 
-    private void checkForExtraProperties(final boolean errorOnExtraProperties, final Class<?> paramClass,
+    private static void checkForExtraProperties(final boolean errorOnExtraProperties, final Class<?> paramClass,
                                          final PObject layer, final String[] extraPropertyToIgnore) {
         final Collection<String> acceptableKeyValues = Sets.newHashSet();
-        for (String name : getAttributeNames(paramClass, FILTER_NON_FINAL_FIELDS)) {
+        for (String name : getAttributeNames(paramClass, FILTER_NON_FINAL_FIELDS::test)) {
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/parser/MapfishParser.java]: What's this ::test?

Comment (pvalsecc) [reply core/src/main/java/org/mapfish/print/parser/MapfishParser.java]: Was a method reference, but not needed. So I've removed it.

**Comment by pvalsecc** [comment]
> Doest it works if we specify a default value?

As far as i can see, no. Should it?

**Comment by pvalsecc** [comment]
Closes #665

**Pull request merged**